test/xpu_api/random/unit_tests/philox_unit_test.pass.cpp

```diff
@@ -0,0 +1,323 @@
+// -*- C++ -*-
+//===-- philox_unit_test.pass.cpp -----------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Unit test for Philox random number generation engine - corner-case testing functions
+
+#include "support/utils.h"
+#include "support/test_config.h"
+
+#include <random>
+#include <oneapi/dpl/random>
+
+namespace ex = oneapi::dpl::experimental;
+
+/* Corner-case testing functions */
+template <typename Engine>
+int api_test();
+
+template <typename Engine>
+int seed_test();
+
+template <typename Engine>
+int discard_test();
+
+template <typename Engine>
+int set_counter_conformance_test();
+
+template <typename Engine>
+int skip_test();
+
+template <typename Engine>
+int counter_overflow_test();
+
+template <typename Engine>
+int discard_overflow_test();
+
+int
+main()
+{
+    int err = 0;
+    
+    std::cout << "void api_test() [Engine = philox4x32]";
+    err += api_test<ex::philox4x32>();
+    std::cout << "void api_test() [Engine = philox4x64]";
+    err += api_test<ex::philox4x64>();
+
+    std::cout << "void seed_test() [Engine = philox4x32]";
+    err += seed_test<ex::philox4x32>();
+    std::cout << "void seed_test() [Engine = philox4x64]";
+    err += seed_test<ex::philox4x64>();
+
+    std::cout << "void discard_test() [Engine = philox4x32]";
+    err += discard_test<ex::philox4x32>();
+    std::cout << "void discard_test() [Engine = philox4x64]";
+    err += discard_test<ex::philox4x64>();
+
+    std::cout << "void set_counter_conformance_test() [Engine = philox4x32]";
+    err += set_counter_conformance_test<ex::philox4x32>();
+    std::cout << "void set_counter_conformance_test() [Engine = philox4x64]";
+    err += set_counter_conformance_test<ex::philox4x64>();
+
+    std::cout << "void skip_test() [Engine = philox4x32]";
+    err += skip_test<ex::philox4x32>();
+    std::cout << "void skip_test() [Engine = philox4x64]";
+    err += skip_test<ex::philox4x64>();
+
+    std::cout << "void counter_overflow_test() [Engine = philox4x32]";
+    err += counter_overflow_test<ex::philox4x32>();
+    std::cout << "void counter_overflow_test() [Engine = philox4x64]";
+    err += counter_overflow_test<ex::philox4x64>();
+
+    std::cout << "void discard_overflow_test() [Engine = philox4x32]";
+    err += discard_overflow_test<ex::philox4x32>();
+    std::cout << "void discard_overflow_test() [Engine = philox4x64]";
+    err += discard_overflow_test<ex::philox4x64>();
+
+    EXPECT_TRUE(!err, "Test FAILED");
+
+    return TestUtils::done(TEST_UNNAMED_LAMBDAS);
+}
+
+/*
+ * All functions below have the following return values:
+ *           0 in case of success
+ *           1 in case of failure
+ */
+
+template <typename Engine>
+int api_test() {
```

Comment:
this one should be covered by engine_methods, may be removed

Reply:
Removed